tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/TensorFlow.java

```diff
@@ -103,6 +102,22 @@ private static OpList libraryOpList(TF_Library handle) {
     }
   }
 
+  public static List<DeviceSpec> listDevices(Optional<DeviceSpec.DeviceType> deviceType, TFE_Context ctx) {
```

Comment:
You shouldn't expose `TFE_Context` at the public level, these classes are generated from the `C API` and we always wrapped them up in public endpoints for more flexibility and scalability. So in this case, I suggest maybe to create an `EagerSession.Context` static nested class that encapsulates a `TFE_Context`?